peps/pep-0750.rst

```diff
@@ -53,837 +33,1089 @@ deferred pending further experience with f-strings. Work on this PEP was
 
 .. code-block:: python
 
-    @runtime_checkable
-    class Decoded(Protocol):
-        def __str__(self) -> str:
-            ...
+    from types import Template, Interpolation
 
-        raw: str
+    def lower_upper(template: Template) -> str:
+        """Render static parts lowercased and interpolations uppercased."""
+        parts: list[str] = []
+        for arg in template.args:        
+            if isinstance(arg, Interpolation):
+                parts.append(str(arg.value).upper())
+            else:
+                parts.append(arg.lower())
+        return "".join(parts)
 
+    name = "world"
+    assert lower_upper(t"HELLO {name}") == "hello WORLD"
 
-These ``Decoded`` objects have access to raw strings. Raw strings are used because tag strings
-are meant to target a variety of DSLs, such as the shell and regexes. Such DSLs have their
-own specific treatment of metacharacters, namely the backslash.
+There is no requirement that template strings are processed in any particular 
+way. Code that processes templates has no obligation to return a string. 
+Template strings are a flexible, general-purpose feature.
 
-However, often the "cooked" string is what is needed, by decoding the string as
-if it were a standard Python string. In the proposed implementation, the decoded object's
-``__new__`` will *store* the raw string and *store and return* the "cooked" string.
+See the `Common Patterns Seen in Processing Templates`_ section for more
+information on how to process template strings. See the `Examples`_ section
+for detailed working examples.
 
-The protocol is marked as ``@runtime_checkable`` to allow structural pattern matching to
-test against the protocol instead of a type. This can incur a small performance penalty.
-Since the ``case`` tests are in user-code tag functions, authors can choose to optimize by
-testing for the implementation type discussed next.
 
-The ``Decoded`` protocol will be available from ``typing``. In CPython, ``Decoded``
-will be implemented in C, but for discussion of this PEP, the following is a compatible
-implementation:
-
-.. code-block:: python
+No Support for Concatenation
+----------------------------
 
-    class DecodedConcrete(str):
-        _raw: str
-
-        def __new__(cls, raw: str):
-            decoded = raw.encode("utf-8").decode("unicode-escape")
-            if decoded == raw:
-                decoded = raw
-            chunk = super().__new__(cls, decoded)
-            chunk._raw = raw
-            return chunk
-
-        @property
-        def raw(self):
-            return self._raw
+Because template strings are designed to be used in custom DSLs, they do not
+support explicit concatenation using ``+``. This is unlike all other string
+literals in Python today. Concatenation may have different semantics in 
+different DSLs or no meaningful semantics at all; accordingly, the ``Template`` 
+type that t-strings evaluate to does not support ``__add__`` or similar methods.
```

Comment:
Elsewhere you talk more broadly than just DSLs -- are we being more restrictive here? I think it's fine just to say that concatenation isn't supported.

Reply:
Agree; fixed.